File: src/test/java/com/dsm/pick/domains/repository/TeacherRepositoryTest.java
package com.dsm.pick.domains.repository;

import com.dsm.pick.domains.domain.Teacher;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;

import javax.persistence.EntityManager;

import static org.junit.Assert.*;

@RunWith(SpringRunner.class)
@DataJpaTest
@ActiveProfiles("dev")
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
class TeacherRepositoryTest {

    @Autowired
    EntityManager entityManager;

    TeacherRepository teacherRepository;

    @BeforeEach
    void beforeEach() {
        teacherRepository = new TeacherRepository(entityManager);
    }

    @Test
    void save_AND_findById() {
        String teacherId = "aaa";
        String teacherPw = "bbb";
        String teacherName = "ccc";

        Teacher teacher = new Teacher();
        teacher.setId(teacherId);
        teacher.setPw(teacherPw);
        teacher.setName(teacherName);

        teacherRepository.save(teacher);

        assertEquals(teacher, teacherRepository.findById(teacherId).get());
    }

    @Test
    void findByRefreshToken() {
        String teacherId = "ddd";
        String teacherPw = "eee";
        String teacherName = "fff";

        Teacher teacher = new Teacher();
        teacher.setId(teacherId);
        teacher.setPw(teacherPw);
        teacher.setName(teacherName);

        teacherRepository.save(teacher);

        String refreshToken = "adfasdf.asdfasd.asdfasdf";
        teacher.setRefreshToken(refreshToken);

        Teacher findTeacher = teacherRepository.findByRefreshToken(refreshToken).get();

        assertEquals(teacher, findTeacher);
    }
}

File: src/test/kotlin/com/dsm/pick/repository/TeacherRepositoryTest.kt
package com.dsm.pick.repository

import org.assertj.core.api.Assertions.*
import org.junit.jupiter.api.Test
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
import org.springframework.test.context.TestConstructor

@DataJpaTest
@TestConstructor(autowireMode = TestConstructor.AutowireMode.ALL)
class TeacherRepositoryTest(
    private val teacherRepository: TeacherRepository
) {

    @Test
    fun `아이디로 선생님 조회 OK`() {
        val teacher = teacherRepository.findTeacherById("teacherId")

        assertThat(teacher?.id).isEqualTo("teacherId")
        assertThat(teacher?.password).isEqualTo("bcc11c9fd8ab1e74e1bc0717239f8f7be24921abca9d5f31705612834dd1d6da3111df3f7120a0e445a91d6c3158b3e09595cd5d06c034c9997d4be2de5a02ca")
        assertThat(teacher?.name).isEqualTo("teacherName")
        assertThat(teacher?.office).isEqualTo("teacherOffice")
    }

    @Test
    fun `없는 아이디로 선생님 조회 OK`() {
        assertThat(teacherRepository.findTeacherById("없는선생님아이디")).isNull()
    }
}